### PR TITLE
New `pr_build.yaml`.

### PR DESCRIPTION
**Pull request by jsfillman**
Updated `pr_build.yaml` to utilize our new reusable workflows:

- All jobs now utilize reusable workflows, located in this repo's `.github/workflows` directory.
- Opening/updating PRs to `develop` cause an image to be built for test, but not pushed.
  - This reduces confusion and excessive images.
- Merging PRs to `develop` causes a new image to be built, pushed, tagged as both `PR-##` and `latest` as `{APPNAME}-develop`.
  - This solves issue with failed tagging, when a PR was merged before original build completes, and reduces tags.
- Opening/updating PRs to `main` or `master` causes an image to be built and immediately pushed with the `PR-##` tag to `{APPNAME}`
- Merging PRs to `main` or `master` causes an image to be built, pushed, and tagged as both `PR-##` and `latest-rc` as `{APPNAME}`.
  - This reduces confusion between merged pre-release PRs and images built with the release workflow.
- All jobs are included in one single file for easy addition via `Actions` > `New workflow`, but skipped jobs are very clear and obvious.

### CODE REVIEW

**Comment by jsfillman** [comment]
To see this in action, and the resulting images, check out https://github.com/kbase/actions_test_repo2

File: workflow-templates/pr_build.yaml [hunk 1]
```diff
@@ -1,47 +1,39 @@
...
-        PR: "${{ github.event.pull_request.number }}"
-        SHA: "${{ github.event.pull_request.head.sha }}"
-        DOCKER_ACTOR: "${{ secrets.GHCR_USERNAME }}"
-        DOCKER_TOKEN: "${{ secrets.GHCR_TOKEN }}"
-      run: "curl https://raw.githubusercontent.com/kbase/.github/main/workflow-templates/scripts/tag_on_pr_merge.sh | bash"
+  build-develop-open:
+    if: github.base_ref == 'develop' && github.event.pull_request.merged == false
+    uses: kbase/.github/.github/workflows/reusable_build.yml@main
+    secrets: inherit
+  build-develop-merge:
```
Comment (bio-boris) [inline workflow-templates/pr_build.yaml:19]: Should the title of this action also contain the word "push"?

File: workflow-templates/pr_build.yaml [hunk 2]
```diff
@@ -1,47 +1,39 @@
...
+    if: github.base_ref == 'develop' && github.event.pull_request.merged == true
+    uses: kbase/.github/.github/workflows/reusable_build-push.yml@main
```
Comment (bio-boris) [inline workflow-templates/pr_build.yaml:21]: Should this file have a mix of underscores and hypens?

File: workflow-templates/pr_build.yaml [hunk 3]
```diff
@@ -1,47 +1,39 @@
...
+    with:
+      name: '${{ github.event.repository.name }}-develop'
+      tags: pr-${{ github.event.number }},latest
+    secrets: inherit
```
Comment (bio-boris) [inline workflow-templates/pr_build.yaml:25]: that's cool!

**Review by bio-boris — COMMENTED** [review]
Couple of minor comments.

File: workflow-templates/pr_build.yaml (hunk 1 above)
Comment (jsfillman) [reply workflow-templates/pr_build.yaml:19]: Resolved this by changing the `name` field to `Pull Request Build, Tag, & Push`.
Keeping filename as `pr_build.yaml` for backwards compatibility.

File: workflow-templates/pr_build.yaml (hunk 3 above)
Comment (jsfillman) [reply workflow-templates/pr_build.yaml:25]: IKR?

**Pull request merged**